src/pose_optimization/pose_optimization.cpp

```diff
@@ -454,100 +145,102 @@ bool Pose_Optimization::compute_pose_with_ransac(const utils::PoseBase& currentP
-        _meanRANSACGetInliersDuration +=
-                (static_cast<double>(cv::getTickCount()) - getRANSACInliersTime) / cv::getTickFrequency();
-
-        // safety
-        if (transformationScore > maxFittingScore)
-        {
-            outputs::log_error("The computed score is higher than the max fitting score");
+        if (canQuit.load())
+            break;
+#endif
+                const double getRandomSubsetStartTime = static_cast<double>(cv::getTickCount());
+                // get a random subset for this iteration
+                const matches_containers::match_container& selectedMatches = get_random_subset(matchedFeatures);
+                _meanGetRandomSubsetDuration +=
+                        (static_cast<double>(cv::getTickCount()) - getRandomSubsetStartTime) / cv::getTickFrequency();
+
+                // compute a new candidate pose to evaluate
+                const double computeOptimisedPoseTime = static_cast<double>(cv::getTickCount());
+                utils::PoseBase candidatePose;
+                if (not Pose_Optimization::compute_optimized_global_pose(currentPose, selectedMatches, candidatePose))
+#ifndef MAKE_DETERMINISTIC
+                    return;
+#else
             continue;
-        }
-
-        // We have a better score than the previous best one
-        if (transformationScore < minScore)
-        {
-            minScore = transformationScore;
-            bestPose = candidatePose;
-            // save features inliers and outliers
-            featureSets.swap(potentialInliersOutliers);
-
-            const double inlierScore =
-                    static_cast<double>(featureSets._point2DSets._inliers.size()) * point2dFeatureScore +
-                    static_cast<double>(featureSets._pointSets._inliers.size()) * pointFeatureScore +
-                    static_cast<double>(featureSets._planeSets._inliers.size()) * planeFeatureScore;
-            if (inlierScore >= enoughInliersScore)
-            {
-                // We can stop here, the optimization is good enough
-                break;
+#endif
+                _meanRANSACPoseOptimizationDuration +=
+                        (static_cast<double>(cv::getTickCount()) - computeOptimisedPoseTime) / cv::getTickFrequency();
+
+                // get inliers and outliers for this transformation
+                const double getRANSACInliersTime = static_cast<double>(cv::getTickCount());
+                matches_containers::match_sets potentialInliersOutliers;
+                const double featureInlierScore =
+                        get_features_inliers_outliers(matchedFeatures, candidatePose, potentialInliersOutliers);
+                _meanRANSACGetInliersDuration +=
+                        (static_cast<double>(cv::getTickCount()) - getRANSACInliersTime) / cv::getTickFrequency();
+
+                // We have a better score than the previous best one, and enough inliers to consider it valid
+                if (featureInlierScore > maxScore)
```

Comment:
This metric is weird. This maximises the number of important features in the result, without taking in consideration the number of inliers or sum of retroprojevtion errors